File: app/src/main/java/net/mbonnin/arcanetracker/HandlesView.java
package net.mbonnin.arcanetracker;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.widget.LinearLayout;

import javax.inject.Inject;

import internal.di.view.AndroidViewInjection;

public class HandlesView extends LinearLayout {
    private ViewManager.Params mParams = new ViewManager.Params();
    private OnTouchListener mListener;

    @Inject ViewManager viewManager;

    public HandlesView(Context context) {
        super(context);
        init();
    }

    public HandlesView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    private void init() {
        AndroidViewInjection.inject(this);
    }

    public void show(boolean show) {
        if (show) {
            viewManager.addView(this, mParams);
        } else {
            viewManager.removeView(this);
        }
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        if (mListener != null) {
            return mListener.onTouch(this, ev);
        }
        return false;
    }

    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        if (mListener != null) {
            return mListener.onTouch(this, ev);
        }
        return false;
    }

    public ViewManager.Params getParams() {
        return mParams;
    }

    public void setListener(OnTouchListener listener) {
        mListener = listener;
    }

    public void update() {
        viewManager.updateView(this, mParams);
    }
}

File: app/src/main/java/net/mbonnin/arcanetracker/HandlesView.kt
package net.mbonnin.arcanetracker

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.view.View
import android.widget.LinearLayout

class HandlesView : LinearLayout {
    val params: ViewManager.Params
    private var mListener: View.OnTouchListener? = null

    constructor(context: Context) : super(context) {
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
    }

    init {
        params = ViewManager.Params()
        clipChildren = false
    }

    fun show(show: Boolean) {
        if (show) {
            ViewManager.get().addView(this, params)
        } else {
            ViewManager.get().removeView(this)
        }
    }

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        return if (mListener != null) {
            mListener!!.onTouch(this, ev)
        } else false
    }

    override fun onTouchEvent(ev: MotionEvent): Boolean {
        return if (mListener != null) {
            mListener!!.onTouch(this, ev)
        } else false
    }

    fun setListener(listener: View.OnTouchListener) {
        mListener = listener
    }

    fun update() {
        ViewManager.get().updateView(this, params)
    }
}
